polygeist/include/mlir/Dialect/Polygeist/Transforms/Passes.td

```diff
@@ -146,6 +146,15 @@ def SCFCanonicalizeFor : Pass<"canonicalize-scf-for"> {
   ];
 }
 
+def SYCLInternalization : InterfacePass<"sycl-internalization", "LoopLikeOpInterface"> {
+  let summary = "Perform local promotion of loop memory accesses";
+  let description = [{
+    Loop tiling is first performed to fit some loop memory accesses in local memory.
```

Comment:
SYCL memory accesses in perfectly nested loops are promoted to shared local memory by leveraging the loop tiling infrastructure.